internal/util/log_reader.go

```diff
@@ -3,40 +3,54 @@ package util
 import (
 	"bufio"
 	"context"
-	"os/exec"
+	"io"
+	"os"
+	"time"
 )
 
 func ReadLog(ctx context.Context, filePath *string, follow bool, c chan []byte, errChan chan error) {
 	if filePath == nil {
+		errChan <- os.ErrInvalid
 		return
 	}
 
-	ctxCancel, cancel := context.WithCancel(ctx)
-	defer cancel()
-
-	tailCmd := exec.CommandContext(ctxCancel, "tail", "-n", "+1")
-	if follow {
-		tailCmd.Args = append(tailCmd.Args, "-f")
-	}
-	tailCmd.Args = append(tailCmd.Args, *filePath)
-
-	reader, err := tailCmd.StdoutPipe()
+	file, err := os.Open(*filePath)
 	if err != nil {
 		errChan <- err
 		return
 	}
-	scanner := bufio.NewScanner(reader)
-	go func() {
-		for scanner.Scan() {
-			c <- scanner.Bytes()
-		}
-	}()
+	defer file.Close()
 
-	err = tailCmd.Start()
-	if err != nil {
-		errChan <- err
-		return
+	// If follow is true, seek to the end of the file
+	if follow {
+		_, err = file.Seek(0, io.SeekEnd)
```

Comment:
I would not seek to the end of the file, the user would just get an empty log then. I would leave it as is and start from the beginning as well.